merlin-framework/src/main/java/gov/nasa/jpl/aerie/merlin/framework/ThreadedTask.java

```diff
@@ -137,15 +142,15 @@ public TaskResponse run(final TaskRequest request) {
             this);
 
         try (final var restore = ThreadedTask.this.rootContext.set(context)) {
-          ThreadedTask.this.task.run();
-          return new TaskResponse.Success(TaskStatus.completed());
+          ThreadedTask.this.returnValue = ThreadedTask.this.task.get();
+          return new TaskResponse.Success(TaskStatus.completed(ThreadedTask.this.returnValue));
         } catch (final TaskAbort ex) {
-          return new TaskResponse.Success(TaskStatus.completed());
+          return new TaskResponse.Success(TaskStatus.completed(VoidEnum.VOID));
```

Comment:
Now that we're providing return values on completion, it looks a little concerning to use `TaskStatus.completed()` when the task is being asked to *abort*. Perhaps we should add a separate `TaskResponse` variant for aborting without a return value.

More specifically, by potentially using either `VoidEnum` or whatever type the task *actually* returns, it becomes much less obvious that we're licensed to *extract* a particular type from the `Success` variant elsewhere